controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -64,6 +65,40 @@ func (in *KubeadmControlPlane) Default() {
 	if !strings.HasPrefix(in.Spec.Version, "v") {
 		in.Spec.Version = "v" + in.Spec.Version
 	}
+
+	ios1 := intstr.FromInt(1)
+	ios0 := intstr.FromInt(0)
+	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
```

Comment:
It probably wouldn't hurt to default here even if in.Spec.RolloutStrategy is nil.

```suggestion
	if in.Spec.RolloutStrategy == nil {
		iin.Spec.RolloutStrategy = RolloutStrategy{}
	}

	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
```

Reply:
I am not against doing this, but do we really need to default here? I think user should define explicitly wanting to use RolloutStrategy?